### PR TITLE
Add Service Update Errors

### PR DESCRIPTION
**Pull request by larioj**
Will be using this errors in cosmos-enterprise for service/update

### CODE REVIEW

File: cosmos-common/src/main/scala/com/mesosphere/cosmos/CosmosError.scala [hunk 1]
```diff
@@ -516,4 +516,47 @@ case class ConversionFromPackageToV2DescribeResponse(
...
+  }
+}
+
+case class ServiceUpdateError(
+  msg: String
+) extends CosmosError {
+  override val getData: Option[JsonObject] = {
+    Some(JsonObject.singleton("msg", msg.asJson))
+  }
+}
```
Comment (jsancio) [inline cosmos-common/src/main/scala/com/mesosphere/cosmos/CosmosError.scala]: For all of these errors, we should include all the contextual information. E.g. what service id did you try to update? What is the old app id vs the new id? etc.

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala [hunk 2]
```diff
@@ -408,7 +408,7 @@ object CosmosApp {
     requirement.fold[EndpointHandler[Req, Res]](new NotConfiguredHandler(operationName))(f)
   }
 
-  private def standardEndpoint[Req, Res](
+  def standardEndpoint[Req, Res](
```
Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala]: @cruhland Can you take a look at this? Is this how you intended for Cosmos to get extended?

Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/Cosmos.scala]: @jsancio yeah, this is fine.

File: cosmos-common/src/main/scala/com/mesosphere/cosmos/CosmosError.scala (hunk 1 above)
Comment (cruhland) [inline cosmos-common/src/main/scala/com/mesosphere/cosmos/CosmosError.scala]: @larioj I noticed you changed all the arguments to `String`. While that will be convenient for the producer of the error, any consumers will not be able to extract contextual information, as @jsancio mentioned. We should have explicit fields for each piece of information, that get surfaced in `getData`.

That said, if you want to merge this as is in the interest of time, that's fine with me. But you'll have to update it later...

**Pull request merged**